core/src/main/java/io/questdb/cairo/RecordChain.java

```diff
@@ -42,6 +42,7 @@ public class RecordChain implements Closeable, RecordCursor, Mutable, RecordSink
     private final MemoryARW mem;
     private final RecordChainRecord recordA = new RecordChainRecord();
     private final RecordChainRecord recordB = new RecordChainRecord();
+    private final RecordChainRecord clone = new RecordChainRecord();
```

Comment:
Would be nice to create this record lazily, i.e. on `cloneRecord` call. We don't want to "pay" for it in queries that don't use `rank()`.